src/catalog/table_catalog.cpp

```diff
@@ -257,6 +257,22 @@ TableCatalogObject::GetColumnObjects(bool cached_only) {
   return column_objects;
 }
 
+/* @brief   get all column objects of this table into cache
+ * @return  map from column name to cached column object
+ */
+std::unordered_map<std::string, std::shared_ptr<ColumnCatalogObject>>
+TableCatalogObject::GetColumnNames(bool cached_only) {
+  if (!valid_column_objects && !cached_only) {
+    auto column_objects = this->GetColumnObjects();
+    std::unordered_map<std::string, std::shared_ptr<ColumnCatalogObject> > column_names;
+    for (auto pair : column_objects) {
```

Comment:
Probably should make this a reference to avoid copy.

Reply:
You mean the loop variable `pair`?